packages/ide/src/client.ts

```diff
@@ -34,6 +34,14 @@ export abstract class IdeClient {
 		this.loadTime = time(2500);
 
 		let appWindow: Window | undefined;
+
+		window.addEventListener("beforeunload", (e) => {
+			const confirmationMessage = "Are you sure you want to navigate away?";
+			(e || window.event).returnValue = confirmationMessage; //IE
+
+			return confirmationMessage; // webkit, safari, chrome
```

Comment:
I don't think the confirmation message is actually used, is it? https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event#Examples